service/controller/v24/adapter/cpf_route_tables.go

```diff
@@ -2,109 +2,118 @@ package adapter
 
 import (
 	"context"
-	"fmt"
-	"strings"
 
-	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
 	"github.com/giantswarm/microerror"
 
 	"github.com/giantswarm/aws-operator/service/controller/v24/controllercontext"
 	"github.com/giantswarm/aws-operator/service/controller/v24/encrypter"
-	"github.com/giantswarm/aws-operator/service/controller/v24/key"
+	"github.com/giantswarm/aws-operator/service/routetable"
 )
 
+type CPFRouteTablesConfig struct {
+	RouteTable *routetable.RouteTable
+
+	AvailabilityZones []v1alpha1.AWSConfigStatusAWSAvailabilityZone
+	EncrypterBackend  string
+	NetworkCIDR       string
+}
+
 type CPFRouteTables struct {
+	routeTable *routetable.RouteTable
+
 	PrivateRoutes []CPFRouteTablesRoute
 	PublicRoutes  []CPFRouteTablesRoute
+
+	availabilityZones []v1alpha1.AWSConfigStatusAWSAvailabilityZone
+	encrypterBackend  string
+	networkCIDR       string
+}
+
+func NewCPFRouteTables(config CPFRouteTablesConfig) (*CPFRouteTables, error) {
+	if config.RouteTable == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.RouteTable must not be empty", config)
+	}
+
+	if config.AvailabilityZones == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.AvailabilityZones must not be empty", config)
+	}
+	if config.EncrypterBackend == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.EncrypterBackend must not be empty", config)
+	}
+	if config.NetworkCIDR == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.NetworkCIDR must not be empty", config)
+	}
+
+	r := &CPFRouteTables{
+		routeTable: config.RouteTable,
+
+		encrypterBackend:  config.EncrypterBackend,
+		availabilityZones: config.AvailabilityZones,
+		networkCIDR:       config.NetworkCIDR,
+	}
+
+	return r, nil
 }
 
-func (a *CPFRouteTables) Adapt(ctx context.Context, config Config) error {
+func (a *CPFRouteTables) Boot(ctx context.Context) error {
```

Comment:
I like the renaming here. I'm fine with the Adapt concept going away. This is clearer and more aligned with rest of our codebase.